contracts/examples/crypto-kitties/kitty-genetic-alg/src/lib.rs

```diff
@@ -0,0 +1,45 @@
+#![no_std]
+
+imports!();
+
+use kitty::{kitty_genes::*, Kitty};
+use random::Random;
+
+#[elrond_wasm_derive::contract(KittyGeneticAlgImpl)]
+pub trait KittyGeneticAlg {
+	#[init]
+	fn init(&self) {}
+
+	// endpoints
+
+	#[endpoint(generateKittyGenes)]
+	fn generate_kitty_genes(&self, matron: Kitty, sire: Kitty) -> SCResult<KittyGenes> {
+		let mut random = Random::new(*self.get_block_random_seed());
+
+		let fur_color_percentage = 1 + random.next_u8() % 99; // val in [1, 100)
+		let matron_fur_color = matron.get_fur_color();
+		let sire_fur_color = sire.get_fur_color();
+		let kitty_fur_color = matron_fur_color.mix_with(
+			&sire_fur_color,
+			fur_color_percentage,
+			100 - fur_color_percentage,
+		);
+
+		let eye_color_percentage = 1 + random.next_u8() % 99; // val in [1, 100)
```

Comment:
Shouldn't we also somehow include the tx nonce into the randomness? Calling generateKittyGenes twice in the same block could produce 2 identical cats. Is that ok?